### PR TITLE
SONARJS-203 Add "sonar.js.ignoreHeaderComments" property

### PR DESCRIPTION
**Pull request by lindamartin**

### CODE REVIEW

File: sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/JavaScriptPlugin.java [hunk 1]
```diff
@@ -95,6 +98,18 @@ public List getExtensions() {
             .subCategory(GENERAL)
             .build(),
 
+      PropertyDefinition.builder(JavaScriptPlugin.IGNORE_HEADER_COMMENTS)
+        .defaultValue(JavaScriptPlugin.IGNORE_HEADER_COMMENTS_DEFAULT_VALUE.toString())
+        .name("jQuery object aliases")
```
Comment (vilchik-elena) [inline sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/JavaScriptPlugin.java]: @lindamartin wat?)

File: sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/JavaScriptPlugin.java [hunk 2]
```diff
@@ -95,6 +98,18 @@ public List getExtensions() {
...
+        .description("If set to \"true\", the file headers (that are usually the same on each file:" +
+          " licensing information for example) are not considered as comments. Thus metrics such as " +
+          "\"Comment lines\" do not get incremented. If set to \"false\", those file headers are " +
+          "considered as comments and metrics such as \"Comment lines\" get incremented.")
```
Comment (vilchik-elena) [inline sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/JavaScriptPlugin.java]: @lindamartin long text. May be it's worth to check it with Ann?

File: sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/JavaScriptPlugin.java [hunk 3]
```diff
@@ -95,6 +98,18 @@ public List getExtensions() {
...
+        .onQualifiers(Qualifiers.MODULE, Qualifiers.PROJECT)
```
Comment (vilchik-elena) [inline sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/JavaScriptPlugin.java]: @lindamartin I don't get why we need `MODULE`. Does JS-plugin support any kind of modules? Or does it mean smth else?

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'src/main/java/org/sonar/javascript/tree' package (org.sonarsource.javascript:javascript-frontend:src/main/java/org/sonar/javascript/tree) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'src/main/java/org/sonar/javascript/tree/impl/statement' package (org.sonarsource.javascript:javascript-frontend:src/main/java/org/sonar/javascript/tree/impl/statement) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1228)

File: sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/JavaScriptPlugin.java (hunk 3 above)
Comment (lindamartin) [inline sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/JavaScriptPlugin.java]: @vilchik-elena in on SonarQube level. For a project it is possible to define modules.

File: sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/JavaScriptPlugin.java (hunk 2 above)
Comment (lindamartin) [inline sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/JavaScriptPlugin.java]: @vilchik-elena checked with Ann and fixed in 2d420480bbed3755ab01a7a3915fc9fc4a0ae36c.

File: sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/JavaScriptPlugin.java (hunk 1 above)
Comment (lindamartin) [inline sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/JavaScriptPlugin.java]: @vilchik-elena copy/paste mistake...fixed in 2d420480bbed3755ab01a7a3915fc9fc4a0ae36c

**Comment by lindamartin** [comment]
@vilchik-elena I'd rather explicitly mention that the property default value is true rather than rely on default behaviour. WDYT ?

**Comment by lindamartin** [comment]
Merged in bbd097e547ff4c08ba91bce60d3f9979c936c98a

**Pull request closed without merging**